code/modules/mob/living/simple_animal/friendly/drone/_drone.dm

```diff
@@ -310,6 +360,34 @@
 		if(cleared)
 			to_chat(src, "--- [class] alarm in [A.name] has been cleared.")
 
+/mob/living/simple_animal/drone/proc/set_shy(new_shy)
+	shy = new_shy
+	shy_update()
+
+/mob/living/simple_animal/drone/proc/shy_update()
+	var/list/drone_bad_areas = make_associative(drone_area_blacklist_flat) + typecacheof(drone_area_blacklist_recursive)
+	var/list/drone_good_machines = make_associative(drone_machinery_whitelist_flat) + typecacheof(drone_machinery_whitelist_recursive)
+	var/list/drone_good_items = make_associative(drone_item_whitelist_flat) + typecacheof(drone_item_whitelist_recursive)
+	if(shy)
+		ADD_TRAIT(src, TRAIT_PACIFISM, DRONE_SHY_TRAIT)
+		LoadComponent(/datum/component/shy, typecacheof(/mob/living/simple_animal/drone), 4, "Your laws prevent this action near %TARGET.", TRUE)
+		LoadComponent(/datum/component/shy_in_room, drone_bad_areas, "Touching anything in %ROOM could break your laws.")
+		LoadComponent(/datum/component/technointrovert, drone_good_machines, "Using %TARGET could break your laws.")
+		LoadComponent(/datum/component/itempicky, drone_good_items, "Using %TARGET could break your laws.")
+	else
+		REMOVE_TRAIT(src, TRAIT_PACIFISM, DRONE_SHY_TRAIT)
+		var/list/qdel_components = list(
+			GetComponent(/datum/component/shy),
+			GetComponent(/datum/component/shy_in_room),
+			GetComponent(/datum/component/technointrovert),
+			GetComponent(/datum/component/itempicky),
+		)
+		for(var/component in qdel_components)
+			qdel(component)
```

Comment:
You're not saving any lines or readability with this list and loop. Just use `RemoveComponent`, which I'm pretty sure exists.

Reply:
`RemoveComponent` is a member of components. This is the way you must do it. I can use `qdel(GetComponent(/datum/component/dead_man_walking))` if you'd like

Reply:
Yes, just qdel.